src/vectors/vectors3f.c

```diff
@@ -0,0 +1,66 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   vectors3f.c                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: tdameros <tdameros@student.42lyon.fr>      +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/08/08 11:47:00 by tdameros          #+#    #+#             */
+/*   Updated: 2023/08/08 11:47:00 by tdameros         ###   ########lyon.fr   */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "libft.h"
+
+#include "vectors.h"
+
+int	vectors3f_initialize(t_vectors3f *vectors3f, size_t size)
+{
+	if (size == 0)
+		size = 1;
+	vectors3f->data = malloc(sizeof(*vectors3f->data) * size);
+	if (vectors3f->data == NULL)
+		return (-1);
+	vectors3f->size = size;
+	vectors3f->length = 0;
+	return (0);
+}
+
+int	vectors3f_add(t_vectors3f *vectors3f, t_vector3f vector3f)
+{
+	t_vector3f	*temp;
+
+	if (vectors3f->length < vectors3f->size)
+		vectors3f->data[vectors3f->length] = vector3f;
+	else
+	{
+		temp = malloc(sizeof(*vectors3f->data) * (vectors3f->size * 2));
+		if (temp == NULL)
+			return (-1);
+		ft_memcpy(temp, vectors3f->data,
+			sizeof(*vectors3f->data) * vectors3f->size);
```

Comment:
replace with `ft_memcpy(temp, vectors3f->data, sizeof(*vectors3f->data) * vectors3f->length);` as they is no point in copying the rest